sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1430,6 +1430,11 @@ class Dataset[T] private[sql](
    */
   @scala.annotation.varargs
   def select(cols: Column*): DataFrame = withPlan {
+    cols.find(_.isInstanceOf[TypedColumn[_, _]]).foreach { typedCol =>
+      throw new AnalysisException(s"$typedCol is a typed column that " +
+        "cannot be passed in untyped `select` API. If you are going to select " +
+        "multiple typed columns, you can use `Dataset.selectUntyped` API.")
+    }
```

Comment:
hmm, we also have such usage in our test:

```scala
df.select(count("*"), countDistinct("*"))
```

So this is still a breaking change? @cloud-fan @HyukjinKwon

Reply:
Ah ... true ..

Reply:
We already shipped `count` as `TypedColumn` for a while, now to change it to `Column` is a breaking change.

In order to not break current usage, I think the fix is to allow feasible typed columns (e.g. `count`) in the untyped `select`.